### PR TITLE
feat: add OrcaRouter as a named AI provider

### PR DESCRIPTION
**Pull request by XiaoHuo888-hue**
## Summary

Add **[OrcaRouter](https://www.orcarouter.ai)** as a named AI provider for commit message generation. OrcaRouter is an OpenAI-compatible gateway, so the new `OrcaRouterEngine` reuses the OpenAI SDK transport via `OpenAiEngine` (mirroring how `DeepseekEngine` does it), forcing the OrcaRouter base URL (`https://api.orcarouter.ai/v1`) by default while still allowing `OCO_API_URL` to override it.

It can also provide optional gateway-level guardrails and firewall policies for prompts, responses, and tool calls. Enabling and configuring those controls is a separate step on the OrcaRouter side: the recommended `balanced` posture is primarily audit-based, while `tight` applies default-deny enforcement. OpenCommit does not configure them; it calls the gateway like any OpenAI-compatible API.

## Changes

- `src/engine/orcarouter.ts` — new `OrcaRouterEngine`, inherits the OpenAI request path and only overrides the base URL + provider name
- `src/engine/openAi.ts` — expose a `providerName` field so subclasses reuse error normalization without re-implementing the request path
- `src/commands/config.ts` — `OCO_AI_PROVIDER_ENUM.ORCAROUTER`, `MODEL_LIST.orcarouter`, default/recommended model `orcarouter/auto`, API-key URL, provider validator
- `src/utils/engine.ts` — engine dispatch
- `src/utils/errors.ts` — billing URL + recommended model
- `src/utils/modelCache.ts` — `fetchOrcaRouterModels` keeps text-output models from the full catalog (direct provider-prefixed models such as `openai/gpt-5` stay selectable)
- `src/commands/setup.ts` — setup provider entry ("OrcaRouter (Smart routing gateway)")
- `README.md` — "Use OrcaRouter as the model gateway" section (security claims qualified)
- `test/unit/orcarouter.test.ts`, `test/unit/modelCache.test.ts`, `test/unit/setup.test.ts` — provider coverage
- `out/cli.cjs` — rebuilt bundle (repo tracks the built output)

## Verification

- `biome check` on changed files: clean
- `prettier --check`: clean
- `tsc --noEmit`: 0 errors
- `npm test`: unit tests pass (incl. new OrcaRouter tests)
- `npm run build`: succeeds; `out/cli.cjs` includes the orcarouter wiring
- L3 live: real key against `https://api.orcarouter.ai/v1/chat/completions` with model `orcarouter/auto` returned **200** with a valid commit message; `GET /v1/models` returned the full catalog including `openai/*`, `anthropic/*`, `google/*`, and `orcarouter/*` aliases

Disclosure: I'm an engineer on the OrcaRouter team.

### CODE REVIEW

File: src/utils/modelCache.ts [hunk 1]
```diff
@@ -184,6 +184,19 @@ export async function fetchDeepSeekModels(apiKey: string): Promise<string[]> {
...
 
+export async function fetchOrcaRouterModels(apiKey: string): Promise<string[]> {
+  return fetchModelList({
+    url: 'https://api.orcarouter.ai/v1/models',
+    headers: { Authorization: `Bearer ${apiKey}` },
+    fallback: MODEL_LIST.orcarouter,
+    mapModels: (data) =>
+      (data as ModelListResponse).data
+        ?.map((model) => model.id)
+        .filter((id) => id.startsWith('orcarouter/'))
```
Comment (di-sukharev) [inline src/utils/modelCache.ts]: The OrcaRouter API intentionally returns provider-prefixed model IDs such as `openai/*`, `anthropic/*`, and `google/*`. Filtering to `orcarouter/*` drops almost the entire catalog, so the setup model picker cannot select the direct models that the gateway exposes. Please either retain all chat-capable models (using the response capability/endpoint metadata where needed), or avoid fetching the full catalog and explicitly limit/document the integration to router aliases. Please add a test showing that a direct model such as `openai/gpt-5` remains selectable if full-catalog support is intended.

File: README.md [hunk 2]
```diff
@@ -258,6 +258,20 @@ oco config set OCO_AI_PROVIDER=ollama OCO_API_KEY=<your_ollama_api_key> OCO_API_
 oco config set OCO_AI_PROVIDER=llamacpp OCO_API_URL=<your_llamacpp_endpoint>
 ```
 
+### Use OrcaRouter as the model gateway
+
+By default OpenCommit uses [OpenAI](https://openai.com).
+
+You could switch to [OrcaRouter](https://www.orcarouter.ai), an OpenAI-compatible gateway that routes every request to the best available model and runs gateway-level, zero-trust security for AI agents on the same endpoint.
```
Comment (di-sukharev) [inline README.md]: This reads as though choosing OrcaRouter automatically enables zero-trust enforcement. According to OrcaRouter's own setup docs, a workspace policy/autonomy level must be applied separately; the recommended `balanced` posture is audit-by-default, while default-deny is the `tight` posture. OpenCommit does not configure any of those settings. Please qualify this as something OrcaRouter *can provide when enabled/configured*, and update the PR summary's default-deny claim as well.

File: src/engine/orcarouter.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  constructor(config: OrcaRouterConfig) {
+    // Call OpenAIEngine constructor with forced OrcaRouter baseURL
+    // Put baseURL first so user config can override it
+    super({
+      baseURL: 'https://api.orcarouter.ai/v1',
+      ...config
+    });
+  }
+
+  // Identical method from OpenAiEngine, re-implemented here
```
Comment (di-sukharev) [inline src/engine/orcarouter.ts]: This method is currently identical to `OpenAiEngine.generateCommitMessage` except for the provider name passed to error normalization. Copying the full request path means future fixes to OpenAI parameters, reasoning handling, or token validation will not automatically reach OrcaRouter. Could we make the provider name a protected field/hook on `OpenAiEngine` and inherit this method, keeping only the base-URL override here?

**Review by di-sukharev — CHANGES_REQUESTED** [review]
Thanks for the thorough implementation and test coverage. The provider wiring is solid and all CI checks pass, but I found two user-facing issues that should be addressed before merge: the model fetch drops nearly the entire OrcaRouter catalog, and the security wording implies enforcement that OpenCommit does not enable. I also left a maintainability note about duplicated OpenAI engine logic. Requesting changes; happy to re-review once these are addressed.

**Comment by XiaoHuo888-hue** [comment]
Thanks for the thorough review — I've addressed all three points and rebased onto the latest `master`.

1. **Full catalog** (`modelCache.ts`): Removed the `orcarouter/*` filter. The gateway returns provider-prefixed IDs such as `openai/gpt-5`, `anthropic/*`, `google/*`, and only 10 `orcarouter/*` aliases out of ~203 models — the filter was dropping ~95% of the catalog. `fetchOrcaRouterModels` now keeps all models returned by `GET /v1/models`. Added a test asserting `openai/gpt-5` (and other provider-prefixed IDs) remain selectable. Verified live: `POST /v1/chat/completions` with model `openai/gpt-5` → 200.

2. **Security wording** (`README.md` + PR summary): Reworded to make clear the zero-trust guards are something OrcaRouter *can provide when configured* — a workspace policy/autonomy level must be applied separately on the OrcaRouter side (the recommended `balanced` posture is audit-by-default, while `tight` is default-deny). OpenCommit does not configure those settings; it just calls the gateway like any OpenAI-compatible API.

3. **Deduplication** (`engine/openAi.ts`, `engine/orcarouter.ts`): `OpenAiEngine` now exposes a `protected providerName` field used for error normalization, and `OrcaRouterEngine` inherits `generateCommitMessage` — it only overrides the base URL and `providerName`. No more duplicated request path, so future OpenAI param/reasoning/token fixes automatically reach OrcaRouter.

Also rebased onto the latest upstream `master` (which moved the provider enum to `src/utils/provider.ts` and added reasoning-model support), regenerated `out/cli.cjs`, and re-ran the full suite: `tsc --noEmit` 0 errors, `npm run lint` clean, 169/169 unit tests pass.

**Review by di-sukharev — APPROVED** [review]
The updated provider integration is ready to merge. The maintainer follow-up covers the remaining factory-path, model-capability, documentation, and regression-test gaps.

**Comment by di-sukharev** [comment]
Thanks again for addressing the review so thoroughly. I added a small maintainer follow-up to cover a couple of edge cases: preserving the default OrcaRouter URL through the real engine factory, filtering non-text models using capability metadata, tightening the documentation wording, and adding regression coverage. The full suite passes with 170 tests, and I've merged the PR into master. Thank you for the contribution!
